pkg/selector/selector_test.go

```diff
@@ -617,12 +635,22 @@ func (p *ec2PricingMock) HydrateSpotCache(days int) error {
 	return p.HydrateSpotCacheErr
 }
 
+func (p *ec2PricingMock) LastOnDemandCacheUTC() *time.Time {
+	return p.lastOnDemandCacheUTC
+}
+
+func (p *ec2PricingMock) LastSpotCacheUTC() *time.Time {
+	return p.lastSpotCacheUTC
+}
+
 func TestFilter_PricePerHour(t *testing.T) {
 	ec2Mock := setupMock(t, describeInstanceTypesPages, "t3_micro.json")
+	now := time.Now()
 	itf := selector.Selector{
 		EC2: ec2Mock,
 		EC2Pricing: &ec2PricingMock{
 			GetOndemandInstanceTypeCostResp: 0.0104,
+			lastOnDemandCacheUTC:            &now,
```

Comment:
☝️ By ensuring the `*CacheUTC` field(s) are populated with a non-`nil` value, the tests exercise relevant code paths in `selector.rawFilter`.